Source/Media/MediaSessionWrapper.swift

```diff
@@ -138,37 +141,22 @@ class MediaSessionWrapper {
     
     // MARK: - Default settings
     private func applyDefaultMediaSettings() {
-        setDefaultFacingMode()
-        setDefaultAudioOutput()
-    }
-    
-    private func setDefaultFacingMode() {
-        let isFront = Phone.sharedInstance.defaultFacingMode == Call.FacingMode.User
+        let isFront = PhoneSettings.defaultFacingMode == Call.FacingMode.User
         mediaSession.setDefaultCamera(isFront)
-    }
-    
-    private func setDefaultAudioOutput() {
-        mediaSession.setDefaultAudioOutput(Phone.sharedInstance.defaultLoudSpeaker)
+        mediaSession.setDefaultAudioOutput(PhoneSettings.defaultLoudSpeaker)
     }
     
     // MARK: - lifecycle
     func prepare(_ mediaOption: MediaOption) {
-        var constraint: MediaConstraint!
-        var localView: MediaRenderView? = nil
-        var remoteView: MediaRenderView? = nil
 
         switch (mediaOption) {
         case .audioOnly:
-            constraint = MediaConstraint(constraint: MediaConstraintFlag.audio.rawValue)
+            mediaSession.mediaConstraint = MediaConstraint(constraint: MediaConstraintFlag.audio.rawValue)
```

Comment:
Okay, sorry I missed this in my first pass. Here's the situation we found. This prepare call is called as kind of a "second initializer" to REALLY initialize the MediaSessionWrapper. It's called only once per Call, either as part of the dial workflow or the answer workflow. Since each Call has it's own MediaSessionWrapper (each of which has a unique new MediaSession) we can be pretty much guaranteed that prepare has no need to reset the MediaSession, but rather is just initializing it once and only once. So I think the changes are good ones in that case, making it a little more clear what's going on, even though it really does not feel very nice that the prepare sits out there as an independent function, looking like it could be called however and whenever you want. It would be nicer if you could only create it completely through an init, and didn't have a multi-part setup process.